src/scripts/galleryTabs.js

```diff
@@ -0,0 +1,288 @@
+      imageSelecter();
+      sliderLength();
+    } else if (productsTabs[3].classList.contains("active")) {
+      topRatedChecker();
+      topRatedChecker(); //funkcja dodana ponownie, by sprawdzic dzialanie na wiekszej ilosci danych, pozniej zostanie usunieta
+      topRatedChecker(); //funkcja dodana ponownie, by sprawdzic dzialanie na wiekszej ilosci danych, pozniej zostanie usunieta
+      topRatedChecker(); //funkcja dodana ponownie, by sprawdzic dzialanie na wiekszej ilosci danych, pozniej zostanie usunieta
+      imageSelecter();
+      sliderLength();
+    };
+  }
+
+  tabChecker();
+
+  function sliderCleaner() {
+    var sliders = slider.querySelectorAll(".slider-image");
+    for (var i = 0; i < sliders.length; i++) {
+      picturesArea.removeChild(sliders[i]);
+    }
+    dataHolder.length = 0;
+  }
+
+  for (var i = 0; i < productsTabs.length; i++) {
+    productsTabs[i].addEventListener("click", function () {
+      for (var j = 0; j < productsTabs.length; j++) {
+        productsTabs[j].classList.remove("active");
+      };
+      this.className += " active";
+    });
+  }
+
+  tabs.addEventListener("click", function () {
+    sliderCleaner();
+    tabChecker();
+  });
+
+  function sliderLength() {
+    var sliders = slider.querySelectorAll(".slider-image");
+    if (sliders.length > 6) {
+      for (var i = 6; i < sliders.length; i++) {
+        sliders[i].style.display = "none";
+      }
+    }
+    pos = 0;
+  }
+
+  function activeSlide(pos) {
+    var sliderImages = slider.querySelectorAll(".js-image");
+    ovlImage.style.backgroundImage = "url('" + dataHolder[pos].src + "')";
+    ovlImage.querySelector(".js-title").innerHTML = dataHolder[pos].name;
+    ovlImage.querySelector(".price-new").innerHTML = dataHolder[pos].newPrice;
+    ovlImage.querySelector(".price-old").innerHTML = dataHolder[pos].oldPrice;
+    for (var j = 0; j < sliderImages.length; j++) {
+      sliderImages[j].className += " js-image-ovl";
+    };
+    sliderImages[pos].classList.remove("js-image-ovl");
+    for (var k = 0; k < sliderStars.length; k++) {
+      sliderStars[k].classList.remove("full");
+    }
+    for (var l = 0; l < dataHolder[pos].stars; l++) {
+      sliderStars[l].className += " full"
+    }
+  }
+
+  function rightClcArr() {
+    var sliders = slider.querySelectorAll(".slider-image");
+    pos += 6;
+    if (pos > sliders.length) {
+      sliderCleaner()
+      tabChecker();
+      sliderLength();
+      pos = 0;
+      tempPos = 0;
+    }
+    for (var i = 0; i < sliders.length; i++) {
+      sliders[i].style.display = "none";
+      tempPos = pos + 6;
+    }
+    for (var j = pos; j < tempPos; j++) {
+
+      if (tempPos > sliders.length) {
+        tempPos = sliders.length
+      }
+      sliders[j].style.display = "flex";
+    }
+  }
+
+  function leftClcArr() {
+    pos -= 6;
+    var slidersLeft = 0;
+    var sliders = slider.querySelectorAll(".slider-image");
+    if (pos < 0) {

```

Comment:
deklaracje zmiennych na samej górze funkcji

Reply:
później, jak kod przejdzie przez kompilator, to i tak wszystkie deklaracje są wywalane na początek funkcji